test/components/plots/linePlotTests.ts

```diff
@@ -242,17 +242,77 @@ describe("Plots", () => {
+        assert.deepEqual(expected.data, actual.data, msg);
+        assert.closeTo(expected.pixelPoints[0].x, actual.pixelPoints[0].x, 0.01, msg);
+        assert.closeTo(expected.pixelPoints[0].y, actual.pixelPoints[0].y, 0.01, msg);
+        assert.deepEqual(expected.selection, actual.selection, msg);
+      }
 
       it("retrieves correct data", () => {
         var dataset3 = [
-          { foo: 0, bar: 1 },
-          { foo: 1, bar: 0.95 }
+          { foo: 0, bar: 0.75 },
+          { foo: 1, bar: 0.25 }
         ];
         linePlot.addDataset("d3", dataset3);
 
-        var lineData = linePlot.getClosestPlotData({x: 490, y: 300});
-        assert.strictEqual(lineData.selection.size(), 1, "only 1 line retreieved");
-        assert.strictEqual(lineData.data[0], dataset3[1], "correct datum retrieved");
+        var lines = d3.selectAll(".line-plot .line");
+        var d0 = dataset3[0];
+        var d0Px = {
+          x: xScale.scale(xAccessor(d0)),
+          y: yScale.scale(yAccessor(d0))
+        };
+
+        var d1 = dataset3[1];
+        var d1Px = {
+          x: xScale.scale(xAccessor(d1)),
+          y: yScale.scale(yAccessor(d1))
+        };
+
+        var expected = {
+          data: [d0],
+          pixelPoints: [d0Px],
+          selection: d3.selectAll([lines[0][2]])
+        };
+
+        var closest = linePlot.getClosestPlotData({x: d0Px.x, y: d0Px.y - 1});
+        assertPlotDataEqual(expected, closest, "if above a point, it is closest");
+
+        closest = linePlot.getClosestPlotData({x: d0Px.x, y: d0Px.y + 1});
+        assertPlotDataEqual(expected, closest, "if below a point, it is closest");
+
+        closest = linePlot.getClosestPlotData({x: d0Px.x + 1, y: d0Px.y + 1});
+        assertPlotDataEqual(expected, closest, "if right of a point, it is closest");
+
+        expected = {
+          data: [d1],
+          pixelPoints: [d1Px],
+          selection: d3.selectAll([lines[0][2]])
+        };
+
+        closest = linePlot.getClosestPlotData({x: d1Px.x - 1, y: d1Px.y});
+        assertPlotDataEqual(expected, closest, "if left of a point, it is closest");
+
+        xScale.domain([0.25,1]);
+        expected.pixelPoints = [{
+          x: xScale.scale(xAccessor(d1)),
+          y: yScale.scale(yAccessor(d1))
+        }];
+
+        closest = linePlot.getClosestPlotData({x: xScale.scale(0.25), y: d1Px.y});
+        assertPlotDataEqual(expected, closest, "only in-view points are considered");
+
+        linePlot = new Plottable.Plot.Line(xScale, yScale);
+        expected = {
+          data: [],
+          pixelPoints: [],
+          selection: d3.selection()
+        };
+
+        closest = linePlot.getClosestPlotData({x: d0Px.x, y: d0Px.y});
+        assert.lengthOf(closest.data, 0, "empty plots return empty data");
+        assert.lengthOf(closest.pixelPoints, 0, "empty plots return empty pixelPoints");
+        assert.isTrue(closest.selection.empty(), "empty plots return empty selection");
```

Comment:
Can you break up this test?  It seems to be testing way too much.

For example "only in-view points are considered" I think should be a test

assert.lengthOf(closest.data, 0, "empty plots return empty data");
+        assert.lengthOf(closest.pixelPoints, 0, "empty plots return empty pixelPoints");
+ assert.isTrue(closest.selection.empty(), "empty plots return empty selection"); should be encapulsated into a single test